### ISSUE TITLE
acl: remove aclAccessorID methods

### ISSUE THREAD

**Issue by dnephin** [hashicorp/consul#12233]
More context in #12166. In that PR we removed two of the `aclAccessorID` methods. 

The two remaining methods are `agent.Agent.aclAccessorID` and `agent/local.Sate.aclAccessorID`. These two are now slightly more expensive because they build a full Authorizer (although that can be cached, so may not be that much more expensive).

In the near future we are planning on improving the ACL permission denied error to include the accessor ID. Once that change is complete we no longer need the `aclAccessorID` methods, because the id they are looking up will already be in the error message. 

This issue is to remind us to remove these extra `aclAccessorID` methods once they are no longer needed.
